### PR TITLE
improve code reuse

### PR DESCRIPTION
**Pull request by bilbeyt**
Close: #1745

### CODE REVIEW

File: beamer/tests/agent/unit/test_handlers.py [hunk 1]
```diff
@@ -25,8 +23,10 @@
 )
 from beamer.tests.agent.utils import make_address
 from beamer.tests.constants import FILL_ID
+from beamer.typing import FillId, Termination
+from beamer.util import get_ERC20_abi
 
-ERC2O_ABI = load_ERC20_abi()
+ERC2O_ABI = get_ERC20_abi()
```
Comment (istankovic) [inline beamer/tests/agent/unit/test_handlers.py:29]: Now that we load the ERC20 abi just once, we can remove this global and instead just call `get_ERC20_abi`.

File: beamer/agent/chain.py [hunk 2]
```diff
@@ -12,15 +12,15 @@
...
+from beamer.agent.relayer import run_relayer_for_tx
 from beamer.agent.state_machine import Context, process_event
-from beamer.agent.typing import BlockNumber, ChainId
-from beamer.agent.util import TransactionFailed, load_ERC20_abi, transact
+from beamer.events import Event, EventFetcher
+from beamer.typing import BlockNumber, ChainId
+from beamer.util import TransactionFailed, get_ERC20_abi, transact
 
-_ERC20_ABI = load_ERC20_abi()
+_ERC20_ABI = get_ERC20_abi()
```
Comment (istankovic) [inline beamer/agent/chain.py:23]: We can get rid of this global now and instead just call `get_ERC20_abi` where needed.

**Comment by istankovic** [comment]
We should also move `contracts_for_web3` from `scripts/_util.py` to `beamer.contracts`.

**Comment by istankovic** [comment]
```
>>> import beamer.agent.relayer as r
>>> r.get_relayer_executable()
PosixPath('XXX/beamer/beamer/agent/data/relayers/relayer-node18-linux-x64')
```

This function returns an incorrect path to the relayer.

**Review by istankovic — CHANGES_REQUESTED** [review]
Overall pretty good, but there are few things to be fixed.

**Comment by bilbeyt** [comment]
> ```
> >>> import beamer.agent.relayer as r
> >>> r.get_relayer_executable()
> PosixPath('XXX/beamer/beamer/agent/data/relayers/relayer-node18-linux-x64')
> ```
> 
> This function returns an incorrect path to the relayer. I wonder how this got past our CI. Do you have an idea?

paid special attention but still missed this good catch :)

**Comment by istankovic** [comment]
> paid special attention but still missed this good catch :)

What is concerning is that the CI did not catch it.
I think we should not fix this until we get the CI to fail first because of this.
(That could require writing a new test, or fixing an existing one, if there is one.)

**Comment by istankovic** [comment]
The Makefile needs updating, we copy relayer binaries to the old location.

**Review by istankovic — APPROVED** [review]
Good work!

**Pull request merged**